src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+    public BookstoreOwner getBookstoreOwner(@RequestParam(value = "id") long id) {
+        return bookstoreOwnerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstoreOwners")
+    public Iterable<BookstoreOwner> getBookstoreOwner() {
+        return bookstoreOwnerRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstoreOwner")
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "name") String name) {
+        BookstoreOwner bookstoreOwner = new BookstoreOwner(name);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstoreOwner;
+    }
+
+
+    //Bookstore REST endpoints
+    @GetMapping("/api/getBookstore")
+    public Bookstore getBookstore(@RequestParam(value = "id") long id) {
+        return bookstoreRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstores")
+    public Iterable<Bookstore> getBookstores() {
+        return bookstoreRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstore")
+    public Bookstore newBookstore(@RequestParam(value = "name") String name, @RequestParam(value = "bookstoreOwnerID") long bookstoreOwnerID) {
+        BookstoreOwner bookstoreOwner = bookstoreOwnerRepository.findById(bookstoreOwnerID);
+        if (bookstoreOwner == null)
+            return null;
+        Bookstore bookstore = new Bookstore(name);
+        bookstore.setBookstoreOwner(bookstoreOwner);
+        bookstoreOwner.addBookstore(bookstore);
+        bookstoreRepository.save(bookstore);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstore;
+    }
+
+
+    //Book REST endpoints
+    @GetMapping("/api/getBook")
+    public Book getBook(@RequestParam(value = "id") long id) {
+        return bookRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBooks")
+    public Iterable<Book> getBooks() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/api/getBooksByBookstore")
+    public Iterable<Book> getBooksByBookstore(@RequestParam(value = "bookstoreID") long bookstoreID) {
+        Bookstore bookstore = bookstoreRepository.findById(bookstoreID);
+        if (bookstore == null)
+            return null;
+        return bookRepository.findByBookstore(bookstore);
+    }
+
+    @PostMapping("/api/newBook")
+    public Book newBook(@RequestParam(value = "name") String name, @RequestParam(value = "isbn") String isbn, @RequestParam(value = "picture") String picture, @RequestParam(value = "description") String description, @RequestParam(value = "author") String author, @RequestParam(value = "publisher") String publisher, @RequestParam(value = "bookstoreID") long bookstoreID) {
+        Bookstore bookstore = bookstoreRepository.findById(bookstoreID);
+        if (bookstore == null)
+            return null;
+        Book book = new Book(name, isbn, picture, description, author, publisher);
+        bookstore.addBook(book);
+        book.setBookstore(bookstore);
+        bookRepository.save(book);
```

Comment:
Wondering whether this save is necessary, does the bookstoreRepository.save() cascade the save to the book?

Reply:
Save not required for Book. 
Addressed in f06ac09992047c1a3eab5bea5cc128da3024d6f4